### DISCUSSION TITLE
Helpers: Warn about unsaved changes

### DISCUSSION THREAD

**Discussion by SeanPM5** [home-assistant/frontend#11239] [Other feature requests]
## The request

When you are editing an **existing** helper on the "Settings" tab, you are not warned when navigating away. 

In this image, I have added three extra options that I haven't saved yet (red box).

![image](https://i.imgur.com/MAYLEJq.png)

If I click the icon in the top right corner (where I pointed the red arrow at), and then navigate BACK to that Settings screen, all of my unsaved changes were lost and no longer there, as you can see in the below screenshot:

![image](https://i.imgur.com/FRHgZxQ.png)

So the feature request is to display some dialog when navigating away like this below. (There already exists a warning dialog when deleting options).

![image](https://i.imgur.com/eTpzPVR.png)

**Comment by anneloredev** [comment]
Hello, I would like to work on this. Is it still available?

**Comment by spacegaier** [comment]
> 
> 
> Hello, I would like to work on this. Is it still available?

Since no one said anything here or is assigned to the issue, this should still be open.

@matthiasdebaat Anything design wise to add or would the target be to simply ensure that the same dialog that is apparently already showing after deleting items is also shown if the user attempts to navigate away via the "Control" button in the top right?

**Comment by spacegaier** [comment]
@anneloreegger I assigned the issue to you now. Please keep us posted. If something comes in the way and you are not able to tackle it after all, let us know as well, so that others can have a go then. Thanks.

**Comment by matthiasdebaat** [comment]
> @matthiasdebaat Anything design wise to add or would the target be to simply ensure that the same dialog that is apparently already showing after deleting items is also shown if the user attempts to navigate away via the "Control" button in the top right?

There are two scenario's a user can navigate away:

### Closing the dialog.
When closing the dialog I would suggest to keep the dialog and use this text:

**Discard changes?**
Your changes will be lost.
Cancel / Discard

_The Discard button would be red_

### Changing the dialog via the control button.
Changing the dialog is a bit more complex. Giving him the discard option is a bad experience:

- If a user want's to discard his changes he can continue.
- If a user want's to keep his changes, he needs to can cancel the dialog, update his changes and reopen the dialog.

I don't have nice solution for this. Make a dialog with the options `Cancel`, `Save` and `Discard`? What do you think?

**Comment by anneloredev** [comment]
> > @matthiasdebaat Anything design wise to add or would the target be to simply ensure that the same dialog that is apparently already showing after deleting items is also shown if the user attempts to navigate away via the "Control" button in the top right?
> 
> There are two scenario's a user can navigate away:
> 
> ### Closing the dialog.
> When closing the dialog I would suggest to keep the dialog and use this text:
> 
> **Discard changes?** Your changes will be lost. Cancel / Discard
> 
> _The Discard button would be red_
> 
> ### Changing the dialog via the control button.
> Changing the dialog is a bit more complex. Giving him the discard option is a bad experience:
> 
> * If a user want's to discard his changes he can continue.
> * If a user want's to keep his changes, he needs to can cancel the dialog, update his changes and reopen the dialog.
> 
> I don't have nice solution for this. Make a dialog with the options `Cancel`, `Save` and `Discard`? What do you think?

I know what you mean. I have experienced similar situations. 
At the moment I cannot think about a better solution, than making a dialog with 3 options, and I also think that this is not a bad solution.

**Comment by anneloredev** [comment]
@spacegaier I unfortunately never really had the time to work on this as I am very busy. So I guess it makes sense to unassign me, so someone else can work on it.
